### PR TITLE
Docker Datadog agent 

### PR DESCRIPTION
**Pull request by arunjadhavt**
for docker datadog container

### CODE REVIEW

**Comment by arunjadhavt** [comment]
Docker class not present in cloud-ops so it was not working there, so have to move here

File: hieradata/puppet_role/ecs.yaml [hunk 1]
```diff
@@ -11,6 +11,9 @@ profile::docker::ecs_agent::running: true
 profile::docker::ecs_agent::cluster_name: "%{::cluster_name}"
 profile::docker::ecs_agent::image: 'amazon/amazon-ecs-agent:v1.17.2'
 
+profile::datadog_docker_agent::running: true
```
Comment (wanix) [inline hieradata/puppet_role/ecs.yaml]: maybe false by default and forced with extrafile

File: site/profile/files/docker_daemon.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#
+# MANAGED BY PUPPET
+#
+
+init_config:
+
+instances:
+  - ## Daemon and system configuration
```
Comment (wanix) [inline site/profile/files/docker_daemon.yaml]: comment inside the array structure for hiera is a bit confusing (you can put the comment after)

File: site/profile/manifests/docker/datadog_docker_agent.pp [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  $image   = 'datadog/docker-dd-agent:12.6.5223',
+  $dd_agent_key = hiera('datadog_agent::api_key')
+
+) {
+
+  require ::profile::docker::host
+  profile::register_profile { 'datadog_docker_agent': }
+
+  docker::run { 'datadog-docker-agent':
+    running => $running,
```
Comment (wanix) [inline site/profile/manifests/docker/datadog_docker_agent.pp]: you may fix the memory usage

File: site/role/manifests/ecs.pp [hunk 4]
```diff
@@ -6,6 +6,9 @@
   require ::profile::docker::host
   require ::profile::docker::registry
   require ::profile::docker::ecs_agent
+  if hiera('profile::datadog_docker_agent::running'){
```
Comment (wanix) [inline site/role/manifests/ecs.pp]: same comment than in puppet-cloud_ops, you should manage this with enablement (also needed for AMI build). You will need to disable it in AMI env file for hiera

Comment (arunjadhavt) [reply site/role/manifests/ecs.pp]: updated for ami build too

File: hieradata/puppet_role/ecs.yaml (hunk 1 above)
Comment (arunjadhavt) [reply hieradata/puppet_role/ecs.yaml]: create PR for hiera file for dev and staging 
https://github.com/Talend/talend-cloud-installer-hiera/pull/88 (Staging)
https://github.com/Talend/talend-cloud-installer-hiera/pull/89 (Integration stack)
https://github.com/Talend/talend-cloud-installer-hiera/pull/90 (QA Stack)

File: site/profile/manifests/docker/datadog_docker_agent.pp [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# docker agent for datadog monitoring
+
+class profile::docker::datadog_docker_agent (
+  $running = true,
+  $image   = 'datadog/docker-dd-agent:12.6.5223',
+
+) {
+
+  require ::profile::docker::host
+  if hiera('profile::datadog_docker_agent::running'){
```
Comment (wanix) [inline site/profile/manifests/docker/datadog_docker_agent.pp]: simply if $running, no need to call hiera() here

Comment (wanix) [reply site/profile/manifests/docker/datadog_docker_agent.pp]: my bad, it's not the same variable, ignore my previous comment

Comment (wanix) [reply site/profile/manifests/docker/datadog_docker_agent.pp]: maybe only precise in comment that profile::datadog_docker_agent is managed by puppet-cloud_ops

**Pull request merged**